Feature/rotate phone v2

Hola-Gordon opened the pull request:
Implements the 2nd version mobile orientation guide that displays once at the start of gameplay. Uses matchMedia API for better practice orientation detection, automatically hides when device is rotated, and includes visual cues for better user experience.

Hola-Gordon committed:
Feat: Use matchMedia for rotate-phone detection

Hola-Gordon committed:
Updated 600px width

halkeye commented on src/objects/OrientationGuide.ts:
```diff
@@ -0,0 +1,145 @@
+            };
+            
+            // Listen for orientation changes
+            mediaQuery.addEventListener('change', orientationChangeHandler);
```
so 3rd parameter of addEventListener, you can do once: and not worry about having to remove it or track if its been seen before, etc

```suggestion
            mediaQuery.addEventListener('change', orientationChangeHandler, { once: true });
```

At least if mediaQuery.addEventListener works like it does for document.addEventListener('

halkeye commented on src/objects/OrientationGuide.ts:
```diff
@@ -0,0 +1,145 @@
+        }
+    }
+
+    public isVisible(): boolean {
```
this isn't actually used anywhere
```suggestion
```

halkeye commented on src/objects/OrientationGuide.ts:
```diff
@@ -0,0 +1,145 @@
+        }
+    }
+
+    private hideOverlay = (): void => {
```
i'm not sure why this is different syntax, for consistentency lets go with
```suggestion
    private hideOverlay(): void {
```
it should be fine cause orientationChangeHandler is an arrow function

Hola-Gordon committed:
Update src/objects/OrientationGuide.ts

Used { once: true } for the orientation change event listener to automatically remove it after triggering

Co-authored-by: Gavin Mogan <github@gavinmogan.com>

Hola-Gordon committed:
Update src/objects/OrientationGuide.ts

Removed the unused isVisible() method

Co-authored-by: Gavin Mogan <github@gavinmogan.com>

Hola-Gordon committed:
Update src/objects/OrientationGuide.ts

Changed hideOverlay method syntax from arrow function to standard method for consistency

Co-authored-by: Gavin Mogan <github@gavinmogan.com>

Hola-Gordon commented:
Thanks for the review! These suggestions make sense and I've implemented all three changes.

halkeye commented on src/objects/OrientationGuide.ts:
```diff
@@ -0,0 +1,144 @@
+                if (!mediaQuery.matches) {
+                    this.hideOverlay();
+                    // Remove the event listener after hiding
+                    mediaQuery.removeEventListener('change', orientationChangeHandler);
```
if you do once, you won't need to remove the listener, cause the listener is no longer there

halkeye commented on src/objects/OrientationGuide.ts:
```diff
@@ -0,0 +1,144 @@
+        }
+    }
+
+        return this.overlay?.visible || false;
```
i guess the suggestion didn't do multiline sorry
it def supported multiple lines at one point

halkeye commented on src/objects/OrientationGuide.ts:
```diff
@@ -0,0 +1,144 @@
+        this.scene = scene;
+        
+        // Only check orientation in the MainMenu scene and only if we haven't shown it before
+        if (scene.scene.key === 'MainMenu' && !OrientationGuide.hasBeenShownBefore) {
```
you probably don't need to check the scene key, you are only adding it to main menu.

Hola-Gordon committed:
Improve OrientationGuide with code review suggestions
